src/CTA.Rules.Analysis/RulesAnalysis.cs

```diff
@@ -246,6 +285,22 @@ private bool AnalyzeChildren(FileActions fileAction, UstList<UstNode> children,
             return containsActions;
         }
 
+        private string GetFullKey(string containingNamespace, string containingClass, string key)
+        {
+            if (string.IsNullOrEmpty(containingNamespace))
+            {
+                return key;
+            }
+            else
+            {
+                if (!string.IsNullOrEmpty(containingClass))
+                {
+                    return $"{containingNamespace}.{containingClass}.{key}";
+                }
+                return $"{containingNamespace}.{key}";
+            }
+        }
```

Comment:
Is it possible to have `$"{containingClass}.{key}"`? If so, one nit/suggestion is to simplify this by using fewer branching blocks:

```csharp
private string GetFullKey(string containingNamespace, string containingClass, string key)
{
	var namespaceToken = string.IsNullOrEmpty(containingNamespace) 
		? string.Empty 
		: $"{containingNamespace}.";

	var classToken = string.IsNullOrEmpty(containingClass) 
		? string.Empty 
		: $"{containingClass}.";

	return $"{namespaceToken}{classToken}{key}";
}
```

Reply:
As discussed, we don't want to use containingClass.key because then the namespace would be empty (meaning semantic analysis failed)